src/persist-txn/src/lib.rs

```diff
@@ -265,11 +265,19 @@ mod proto {
 /// Adds the full set of all persist-txn `Config`s.
 pub fn all_dyncfgs(configs: ConfigSet) -> ConfigSet {
     configs
+        .add(&crate::INIT_FORGET_ALL)
+        .add(&crate::operator::DATA_SHARD_RETRYER_CLAMP)
         .add(&crate::operator::DATA_SHARD_RETRYER_INITIAL_BACKOFF)
         .add(&crate::operator::DATA_SHARD_RETRYER_MULTIPLIER)
-        .add(&crate::operator::DATA_SHARD_RETRYER_CLAMP)
 }
 
+/// Used by storage controller as a temporary escape hatch for #25992.
+pub const INIT_FORGET_ALL: Config<bool> = Config::new(
+    "persist_txn_init_forget_all",
+    false,
```

Comment:
Shouldn't this be true so that the default behavior is that we call `forget_all`?

Reply:
I was thinking that we don't strictly need the forget_all anymore (and with the bug it's accumulating things each week), plus if some envd got into a crash loop, we'd have to react by flipping the LD. Open to flipping it though!

(NB: The second commit changes the default from true->false, and if we cherry-picked this, we'd only do the first one. If that changes things for you.)

Reply:
Oh got it, I misunderstood and didn't realize there were two commits. This LGTM as is.